### PR TITLE
[SPARK-50174][SQL] Factor out `UnresolvedCatalogRelation` resolution

### PR DESCRIPTION
**Pull request by vladimirg-db**
### What changes were proposed in this pull request?

Factor out `UnresolvedCatalogRelation` resolution into a separate `FindDataSourceTable` method to reuse it in the single-pass Analyzer.

### Why are the changes needed?

Context: https://issues.apache.org/jira/browse/SPARK-49834

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### Was this patch authored or co-authored using generative AI tooling?

copilot.vim.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala [hunk 1]
```diff
@@ -320,6 +313,20 @@ class FindDataSourceTable(sparkSession: SparkSession) extends Rule[LogicalPlan]
         v1Relation
       }
   }
+
+  def resolveUnresolvedCatalogRelation(
+      unresolvedCatalogRelation: UnresolvedCatalogRelation): LogicalPlan =
+    unresolvedCatalogRelation match {
```
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala:319]: Let's put the method body to {}, to follow conventions of the style guide https://github.com/databricks/scala-style-guide?tab=readme-ov-file#curly-braces

**Review by MaxGekk — COMMENTED** [review]
lgtm except of a minor comment.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala (hunk 1 above)
Comment (vladimirg-db) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala:319]: Done, thanks

**Review by MaxGekk — APPROVED** [review]

**Comment by vladimirg-db** [comment]
@MaxGekk tests passed, one unrelated suite failed
image

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @vladimirg-db.

**Pull request closed without merging**